Adapt map layout buttons to tablet - #12 issue

vasukiarunachalam opened the pull request:
Layout-land folder added for gamemap.xml, to make the layouts work in tablets

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,49 @@
+        android:id="@+id/HouseButton"
+        android:layout_width="@dimen/map_click_width"
+        android:layout_height="@dimen/map_click_height"
+        android:layout_marginLeft="57dp"
```
Don't use raw values. Define them on dimens.xml and reference them here using `@dimen/dimension_name`

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,49 @@
+        android:id="@+id/HospitalButton"
+        android:layout_width="@dimen/map_click_width"
+        android:layout_height="@dimen/map_click_height"
+        android:layout_marginRight="59dp"
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,49 @@
+        android:layout_alignParentTop="true"
+        android:layout_alignParentRight="true"
+        android:layout_alignParentEnd="true"
+        android:layout_marginEnd="59dp" />
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,49 @@
+        android:alpha="0.00"
+        android:text="@string/school"
+        android:layout_alignParentBottom="true"
+        android:layout_alignLeft="@+id/HospitalButton"
```
Use `@+id` only when defining a new id in the `android:id` field. If you are referring an existing id, use `@id`

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,49 @@
+        android:alpha="0.00"
+        android:text="@string/boyfriend"
+        android:layout_alignParentBottom="true"
+        android:layout_alignLeft="@+id/HouseButton"
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,49 @@
+        android:text="@string/boyfriend"
+        android:layout_alignParentBottom="true"
+        android:layout_alignLeft="@+id/HouseButton"
+        android:layout_alignStart="@+id/HouseButton" />
```
Same

alma-castant commented:
I left some comments on the code. What are the changes between the other map layout and this one? If it was only the margin values, you could instead create a new dimens.xml file where you redefine these margins for the landscape mode.

After you make your changes, add your files and commit doing `git commit --amend` so you edit the current commit. Then push and the PR will be automatically updated. Please leave a comment so I know you updated it (otherwise I will not be notified)

vasukiarunachalam commented:
I changed the layout as per your comments. In landscape mode the places look more wider and so I defined the size of buttons in dimens.xml. Also defined the margin values. I also changed the id as you had commented. Please let me know if it is correct.

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+    android:background="@drawable/map">
+
+    <Button
+        android:id="@id/HouseButton"
```
When you define a new id in the `android:id` field you should use `@+id`. Use `@id` elsewhere where you just reference an already-existing id

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+        android:text="@string/house" />
+
+    <Button
+        android:id="@id/HospitalButton"
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+        android:text="@string/hospital" />
+
+    <Button
+        android:id="@id/SchoolButton"
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+        />
+
+    <Button
+        android:id="@id/BoyfriendButton"
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+        android:layout_height="@dimen/land_map_click_height"
+        android:layout_marginLeft="@dimen/land_house_button_margin_left"
+        android:layout_marginTop="@dimen/land_house_button_margin_top"
+        android:alpha="0.00"
```
Can you move the alpha value to `dimens.xml` :) ?

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+        android:layout_height="@dimen/land_map_click_height"
+        android:layout_marginLeft="@dimen/land_hospital_button_margin_left"
+        android:layout_marginTop="@dimen/land_hospital_button_margin_top"
+        android:alpha="0.00"
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+        android:layout_height="@dimen/land_map_click_height"
+        android:layout_marginLeft="@dimen/land_school_button_margin_left"
+        android:layout_marginTop="@dimen/land_school_button_margin_top"
+        android:alpha="0.00"
```
Same

alma-castant commented on PowerUp/app/src/main/res/layout-land/gamemap.xml:
```diff
@@ -0,0 +1,45 @@
+        android:layout_height="@dimen/land_map_click_height"
+        android:layout_marginLeft="@dimen/land_boyfriend_button_margin_left"
+        android:layout_marginTop="@dimen/land_boyfriend_button_margin_top"
+        android:alpha="0.00"
```
Same

alma-castant commented:
I left some more comments, but I am nor sure you considered what I said before: Is there any change between the vertical and the horizontal map layouts besides the dimensions? In case there is not, you should then make two versions of the dimensions file rather than two versions of the layout.
You also need to resolve conflicts. Pull the latest changes from the repo and resolve the conflicts in your computer. You currently have 3 commits, so I would recommend to create a new PR with one commit only if possible

vasukiarunachalam commented:
Thank you, I understand that we can define different  "dimens.xml" for single gamemap layout. But the normal gamemap.xml has few definitions for alignment (eg alignbaseline of house button) with other buttons that does not fit for landscape mode. The buttons become more cluttered. If I can change the gamemap.xml and make it most common one with only button sizes and margins defined in different dimens.xml, then I think it will work for all sizes. Shall I do it ? 

Currently without any of my commits -
For smaller phones-
1. The button size is 100 * 100 dp and looks good in portrait. When we look it in landscape mode, the buttons look huge when compared to background map places and are aligned in wrong places. We need to define 50 *150 dp button size in landscape.

For Tablets 7' and 10' -
2. The button sizes 100 * 100 dp are too small for background map places. Both in portrait and landscape the buttons are too small. We need to define 100 * 200 dp button sizes.

So I think we need to define dimens.xml for -
values
values-land
values-w820dp ( This is already there in our repo)
values-w820dp-land

Please let me know if my thoughts are correct and I can proceed with changes.

alma-castant commented:
I don't really understand your first paragraph. If, on top of the dimensions, you need to change the alignments, it's ok to have game_layout defined twice. Adapting the values for different screen sizes is also a great idea

vasukiarunachalam commented:
Sorry if my explanation was confusing. I think we need not define a layout-land/gamemap.xml. Instead we can edit the gamemap.xml to make it more generalized only with button size and margin values like-

 <Button
+        android:id="@id/SchoolButton"
+        android:layout_width="@dimen/map_click_width"
+        android:layout_height="@dimen/map_click_height"
+        android:layout_marginLeft="@dimen/lschool_button_margin_left"
+        android:layout_marginTop="@dimen/school_button_margin_top"
+        android:alpha="0.00"
+        android:text="@string/school"
+        />

But currently we have -
<Button
        android:id="@+id/SchoolButton"
        android:layout_width="@dimen/map_click_width"
        android:layout_height="@dimen/map_click_height"
        android:layout_alignLeft="@id/HospitalButton"
        android:layout_alignParentBottom="true"
        android:layout_alignStart="@id/HospitalButton"
        android:layout_marginBottom="@dimen/school_button_margin_bottom"
        android:alpha="0.00"
        android:text="@string/school" />

The alignment definitions in line 4,5,6 make the buttons constrained. So shall I change the gamemap.xml and also define different dimens.xml in value,value-land,values-w820dp folders.

alma-castant commented:
Feel free to change the dimens files only if that works well. I think it's better if you submit your new code on a new PR so I can look at it, test it and give better feedback ;)
